apps/omg_eth/lib/eth.ex

```diff
@@ -85,35 +55,12 @@ defmodule OMG.Eth do
   @spec zero_address() :: address()
   def zero_address(), do: <<0::160>>
 
-  def call_contract(contract, signature, args, return_types) do
-    data = signature |> ABI.encode(args)
-
-    with {:ok, return} <- Ethereumex.HttpClient.eth_call(%{to: to_hex(contract), data: to_hex(data)}),
-         do: decode_answer(return, return_types)
-  end
-
-  defp decode_answer(enc_return, return_types) do
-    enc_return
-    |> from_hex()
-    |> ABI.TypeDecoder.decode(return_types)
-    |> case do
-      [single_return] -> {:ok, single_return}
-      other when is_list(other) -> {:ok, List.to_tuple(other)}
-    end
-  end
-
-  @spec submit_block(
-          binary(),
-          pos_integer(),
-          pos_integer(),
-          RootChain.optional_address_t(),
-          RootChain.optional_address_t()
-        ) ::
-          {:error, binary() | atom() | map()} | {:ok, binary()}
-  def submit_block(hash, nonce, gas_price, from \\ nil, contract \\ %{}) do
-    contract = Config.maybe_fetch_addr!(contract, :plasma_framework)
-    from = from || from_hex(Application.fetch_env!(:omg_eth, :authority_addr))
-    backend = Application.fetch_env!(:omg_eth, :eth_node)
+  @spec submit_block(binary(), pos_integer(), pos_integer()) ::
+          {:error, binary() | atom() | map()} | {:ok, <<_::256>>}
+  def submit_block(hash, nonce, gas_price) do
```

Comment:
removing all the optional paramters here, unifying access to contracts and configuration